[yyihaoc] iP

yyihaoc opened the pull request:
### Astor

> "A cluttered mind is a manifestation of a cluttered life" 

Astor is here to help you record your tasks and free your mind. It's

- text-based and graphic-based
- easy to learn
- fast

and **most importantly** ...

IT'S FREE and HAPPY to help you remember as many stuff as you want 😆


### All you need to do is,

1. Download it from [here](https://github.com/yyihaoc/ip)
2. Run it
3. Type anything to view the guide for recording tasks
4. Be happier and less stressed out about forgetting it  🤗

### Features

- [x] Manages (add, delete, mark, list) a wide range of tasks (TODO, DEADLINE, EVENT)
- [x] Find relevant tasks with a keyword
- [x] Works with text based input and graphical interface
- [ ] Allows you to view upcoming tasks in chronological order _(Coming soon!)_
- [ ] More interactive graphical interface 

If you are a Java programmer, you can use it to revise your understanding of Java too. Here's the run method that shows how the application works!

```Java
public void run() {
        ui.showWelcome();
        boolean isExit = false;
        while (!isExit) {
            try {
                String fullCommand = ui.readCommand();
                Command c = Parser.process(fullCommand);
                c.execute(taskList, ui, storage);
                isExit = c.isExit();
            } catch (AstorException e) {
                ui.showError(e.getMessage());
            } catch (IOException e) {
                ui.showError("An error occurred while writing to the file.");
            } finally {
                ui.showLine();
            }
        }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yyihaoc committed:
Updated my chatbot new name and response

yyihaoc committed:
Made Astor echo user inputs and terminates with bye

yyihaoc committed:
Added list to store user inputs

yyihaoc committed:
Added functions to allow task to be marked as done or not, handled exceptions to inputs

yyihaoc committed:
Added subclasses Todo, Deadline, Event to track specific tasks, added functionality for tracking each of those unique tasks

yyihaoc committed:
Removed blankspace before next line in Astor.java, and semi-automate the testing of text UIs with sample inputs and expected outputs

yyihaoc committed:
Handled exceptions and reorganised Astor.java

yyihaoc committed:
Handled exception, reorganised Astor.java and updated expected output

yyihaoc committed:
Added functionality to delete tasks, updated testing inputs

yyihaoc committed:
Added enums for clarity and siplication

yyihaoc committed:
Add DataFileModifier.java

Tasks cannot be recorded to local hard drive.

Let's,
* add a class to interact with text file that
  stores previous data
* change the main file to get the data from
  text file before receiving user inputs

yyihaoc committed:
Merge branch 'branch-Level-7'

Allow users to store previous recorded tasks

yyihaoc committed:
Allow Astor to understand date and time

yyihaoc committed:
Merge branch 'branch-Level-8', allow data and time

yyihaoc committed:
Refactor code from Astor.java to Ui.java

yyihaoc committed:
Rename and refactor Storage.java

yyihaoc committed:
Refactor with storage, ui, tasklist, parser

yyihaoc committed:
Merge branch 'branch-A-MoreOOP'

yyihaoc committed:
Organise files into packages

yyihaoc committed:
Update test input and output

yyihaoc committed:
Merge branch 'branch-A-Gradle'

yyihaoc committed:
Add gradle

yyihaoc committed:
Add JUnit tests for TodoCommand, TaskList

yyihaoc committed:
Create JAR File

yyihaoc committed:
Add JavaDoc for most of the clases

yyihaoc committed:
Add find feature to search for matching tasks

yyihaoc committed:
Merge branch 'branch-A-JavaDoc', includes JavaDoc

yyihaoc committed:
Add find feature to search tasks

ethantyh692 commented on src/main/java/astor/Astor.java:
```diff
@@ -0,0 +1,72 @@
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```
I like how the boolean variable is named like a boolean.

ethantyh692 commented on src/main/java/astor/Storage.java:
```diff
@@ -0,0 +1,137 @@
+ */
+public class Storage {
+
+    private FileWriter fw;
```
Perhaps this could be more explicitly spelt out? fw may not be clear later in the code

ethantyh692 commented on src/main/java/astor/TaskList.java:
```diff
@@ -0,0 +1,147 @@
+     * @return reply of the chatbot after marking tasks as done
+     * @throws MarkTaskOutOfRangeException if an invalid taskIndex is given by user
+     */
+    public String unMarkTaskDone(int taskIndex, Storage storage) throws MarkTaskOutOfRangeException {
```
Perhaps unMark should be unmark since unmark is a single word?

ethantyh692 commented on src/main/java/astor/exception/DeleteTaskNumberException.java:
```diff
@@ -0,0 +1,12 @@
+ *
+ * @author Choi Yi Hao
+ */
+public class DeleteTaskNumberException extends AstorException {
```
I like the clear naming of exceptions

ethantyh692 commented on src/main/java/astor/Astor.java:
```diff
@@ -0,0 +1,72 @@
+package astor;
```
Namings were largely well done. Good job!

ethantyh692 reviewed (commented):
LGTM. Just a few nits and suggestions.

chrisjohntan commented on src/main/java/astor/task/Event.java:
```diff
@@ -0,0 +1,75 @@
+ * This class extends the Task class and includes both a start and an end time for the event.
+ */
+public class Event extends Task {
+    private final LocalDateTime start;
+    private final LocalDateTime end;
```
Could these constants be uppercased?

chrisjohntan commented on src/main/java/astor/task/Task.java:
```diff
@@ -0,0 +1,104 @@
+ * and retrieve its description.
+ */
+public class Task {
+    private boolean status;
```
Could this be named to sound more like a boolean?
```suggestion
    private boolean isDone;
```

chrisjohntan commented on src/main/java/astor/task/Todo.java:
```diff
@@ -0,0 +1,46 @@
+        super(taskInfo);
+    }
+
+    /**
+     * Returns a string representation of the to-do task.
+     *
+     * The string representation includes the task type, completion status, and task description.
+     * The status is shown as "[X]" if the task is done, and "[ ]" if it is not.
+     *
+     * @return a string representation of the to-do task
+     */
+    @Override
+    public String toString() {
+        String s = "[T] ";
+        if (this.isDone()) {
+            s += "[X] ";
+        } else {
+            s += "[ ] ";
+        }
+        s += this.getTaskInfo();
+        return s;
+    }
+
+    /**
+     * Returns a string description of the to-do task data for storage.
+     *
+     * The string includes the task type, completion status, and task description.
+     *
+     * @return a string description of the to-do task data
+     */
+    @Override
```
I like your well detailed header comments 👍

chrisjohntan reviewed (commented):
LGTM. Just a few nits to fix

yyihaoc committed:
Add GUI and retain text-based UI

yyihaoc committed:
Merge branch 'branch-Level-10', add GUI

yyihaoc committed:
Add checkstyle

yyihaoc committed:
Merge branch 'branch-A-CheckStyle'

yyihaoc committed:
Allow search using varargs, fix bugs for testcases

Find command previously only accepts one keyword to search for tasks.

Searching using multiple keywords could be useful because users might
need to search for several related tasks at the same time.

Use varargs and loop through the search keywords provided to
check for matching tasks, and returns them. Use stream functions for
better readability.

yyihaoc committed:
Merge branch 'branch-A-Varargs'

yyihaoc committed:
Include assert statements to verify assumptions

Assert statements make program more robust and less prone
to errors.

Include assert statements mainly to test if arguments to
methods are null, which they are not supposed to be.

yyihaoc committed:
Merge pull request #2 from yyihaoc/branch-A-Assertions

Include assert statements to verify assumptions

yyihaoc committed:
Improve code quality by restructuring code blocks

Repeated code across sibling classes. Magic literals
may cause confusion.

Assign magic literals to variables named for easy
understanding. Refactor code to remove duplication and
allow for the same logic to be found in the same class

TaskList.java: Create all the output strings and returns
them instead of having some in method execute under
Command classes

yyihaoc committed:
TaskList.java: Allow streams to search tasks

yyihaoc committed:
Merge pull request #3 from yyihaoc/branch-A-CodeQuality

Improve code quality by restructuring code blocks

yyihaoc committed:
Merge branch 'master' into branch-A-CodeQuality

yyihaoc committed:
Merge from master

yyihaoc committed:
Revise updates from other branches

yyihaoc committed:
Merge pull request #4 from yyihaoc/branch-A-Streams

Revise updates from other branches, add streams

yyihaoc committed:
Prevent duplicate records of tasks

yyihaoc committed:
Merge pull request #5 from yyihaoc/branch-C-DetectDuplicates

Prevent duplicate records of tasks

yyihaoc committed:
Set up CI

yyihaoc committed:
Resolve style issues

yyihaoc committed:
Merge branch 'branch-StyleEdit'

yyihaoc committed:
Delete I/O redirection tests

yyihaoc committed:
Remove Windows for code testing

yyihaoc committed:
Add introduction message to GUI

yyihaoc committed:
Resolve warnings, add time format exceptions

yyihaoc committed:
Add windows-latest in platform for CI

yyihaoc committed:
Merge branch 'branch-RefineGui'

yyihaoc committed:
TimeFormatException.java: Add Javadocs

yyihaoc committed:
Add normalise line endings for window CI

yyihaoc committed:
Add normalise line endings for window CI

yyihaoc committed:
Refine GUI to improve UX

yyihaoc committed:
Merge branch 'branch-A-BetterGui'

yyihaoc committed:
Modify the return message for personality

yyihaoc committed:
Merge branch 'branch-A-Personality'

yyihaoc committed:
Add product representative screenshot

yyihaoc committed:
Deadline.java: Remove whitespace

yyihaoc committed:
READ.md: Add introductions and instructions

yyihaoc committed:
Merge branch 'branch-A-UserGuide'

yyihaoc committed:
Fix data path for storage

yyihaoc committed:
Add product screenshot to READ.md
